akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/BehaviorBuilder.scala

```diff
@@ -177,23 +180,25 @@ object BehaviorBuilder {
  * INTERNAL API
  */
 @InternalApi
-private final class BuiltBehavior[T](messageHandlers: List[Case[T, T]], signalHandlers: List[Case[T, Signal]])
+private final class BuiltBehavior[T](messageHandlers: Array[Case[T, T]], signalHandlers: Array[Case[T, Signal]])
     extends ExtensibleBehavior[T] {
 
-  override def receive(ctx: TypedActorContext[T], msg: T): Behavior[T] = receive(msg, messageHandlers)
+  override def receive(ctx: TypedActorContext[T], msg: T): Behavior[T] = receive(msg, messageHandlers, 0)
 
-  override def receiveSignal(ctx: TypedActorContext[T], msg: Signal): Behavior[T] =
-    receive(msg, signalHandlers)
+  override def receiveSignal(ctx: TypedActorContext[T], msg: Signal): Behavior[T] = receive(msg, signalHandlers, 0)
 
   @tailrec
-  private def receive[M](msg: M, handlers: List[Case[T, M]]): Behavior[T] =
-    handlers match {
-      case Case(cls, predicate, handler) :: tail =>
-        if ((cls.isEmpty || cls.get.isAssignableFrom(msg.getClass)) && (predicate.isEmpty || predicate.get.apply(msg)))
-          handler(msg)
-        else receive(msg, tail)
-      case Nil =>
+  private def receive[M](msg: M, handlers: Array[Case[T, M]], idx: Int): Behavior[T] = {
+    if (handlers.length == 0) {
+      Behaviors.empty[T]
```

Comment:
wasn't this changed from `Behaviors.unhandled[T]`? why?

Reply:
it was `Behaviors.unhandled[T]` but  @johanandren asked to change to `Behaviors.empty` even them are same

Reply:
![image](https://user-images.githubusercontent.com/501740/165218235-2867c452-0d76-46f4-90db-9c2eb2053c2b.png)

Reply:
from a semantics point of view, It should be `empty` when there is no handlers and `unhandled` when handlers is not empty but not any match

Reply:
I don't think they are exactly the same. There is definitely code that is checking for UnhandledBehavior. I don't know if that would be related here or not, but I would rather not risk introducing a regression.

Reply:
Cool, let me first revert it back then.